### PR TITLE
Feature/rqmupdate

### PR DESCRIPTION
**Pull request by prashelke**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is point to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [x] Sanity Tests are successfully executed for Existing Functionality
- [x] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [x] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments

### CODE REVIEW

**Comment by prashelke** [comment]
Unit test is added in Rqm Repository we are not able to added unit test in ginger because we only used Rqm Repository dll in ginger so need to add into Regression suite

File: Ginger/Ginger/App.config [hunk 1]
```diff
@@ -1,7 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <configuration>
-  <configSections>
-    </configSections>  
+	<configSections>
+	</configSections>
+	<appSettings>
+		<add key="TraceActivateSwitch" value="True"/>
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.config:6]: do we need this settings ?

File: Ginger/Ginger/App.xaml.cs [hunk 2]
```diff
@@ -242,6 +244,13 @@ private void Application_Startup(object sender, StartupEventArgs e)
         {            
             Amdocs.Ginger.CoreNET.log4netLib.GingerLog.InitLog4Net();
 
+            if (System.Diagnostics.Trace.Listeners.Contains(CustomTraceListener.Instance) == false)
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.xaml.cs:247]: Use Ginger reporter log for logging

File: Ginger/Ginger/CustomTraceListener.cs [hunk 3]
```diff
@@ -0,0 +1,572 @@
+#region License
```
Comment (Maheshkale447) [inline Ginger/Ginger/CustomTraceListener.cs:1]: Use Ginger logs for logging

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 4]
```diff
@@ -9209,15 +9209,15 @@ public string GetCurrentPageSourceString()
...
             {
                 options.PageLoadStrategy = OpenQA.Selenium.PageLoadStrategy.Normal;
             }
-            else if (PageLoadStrategy.ToLower() == nameof(OpenQA.Selenium.PageLoadStrategy.Eager).ToLower())
+            else if (PageLoadStrategy != null && PageLoadStrategy.ToLower() == nameof(OpenQA.Selenium.PageLoadStrategy.Eager).ToLower())
             {
                 options.PageLoadStrategy = OpenQA.Selenium.PageLoadStrategy.Eager;
             }
-            else if (PageLoadStrategy.ToLower() == nameof(OpenQA.Selenium.PageLoadStrategy.None).ToLower())
+            else if (PageLoadStrategy != null && PageLoadStrategy.ToLower() == nameof(OpenQA.Selenium.PageLoadStrategy.None).ToLower())
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:9220]: Please check what PageLoadStrategy is coming null and if it is unavoidable, add single condition to handle null PageLoadStrategy .

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/App.config (hunk 1 above)
Comment (prashelke) [reply Ginger/Ginger/App.config:6]: yes for Alm Repository Logs

File: Ginger/Ginger/App.xaml.cs (hunk 2 above)
Comment (prashelke) [reply Ginger/Ginger/App.xaml.cs:247]: this is for alm repository listener is up or not

File: Ginger/Ginger/CustomTraceListener.cs (hunk 3 above)
Comment (prashelke) [reply Ginger/Ginger/CustomTraceListener.cs:1]: this is for alm repository listener is up or not

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs (hunk 4 above)
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:9220]: Done

**Comment by prashelke** [comment]
NEw PR Created For the Same

**Pull request closed without merging**